internal/status/advl4_status.go

```diff
@@ -40,48 +40,63 @@ type UpdateGWStatusConditionOptions struct {
 	Reason  string               // reason for transition
 }
 
-// TODO: handle bulk during bootup
 func UpdateGatewayStatusAddress(options []UpdateStatusOptions, bulk bool) {
-	for _, option := range options {
-		gatewayNSName := strings.Split(option.ServiceMetadata.Gateway, "/")
-		gw, err := lib.GetAdvL4Clientset().NetworkingV1alpha1pre1().Gateways(gatewayNSName[0]).Get(gatewayNSName[1], metav1.GetOptions{})
-		if err != nil {
+	gatewaysToUpdate, updateGWOptions := parseOptionsFromMetadata(options, bulk)
+	var updateServiceOptions []UpdateStatusOptions
+
+	// gatewayMap: {ns/gateway: gatewayObj}
+	// this pre-fetches all gateways to be candidates for status update
+	// after pre-fetching, if a status update comes for that gateway, then the pre-fetched gateway would be stale
+	// in which case gateway will be fetched again in updateObject, as part of a retry
+	gatewayMap := getGateways(gatewaysToUpdate, bulk)
+	for _, option := range updateGWOptions {
+		if gw := gatewayMap[option.IngSvc]; gw != nil {
+			updateServiceOptions = append(updateServiceOptions, UpdateStatusOptions{
+				Vip: option.Vip,
+				Key: option.Key,
+				ServiceMetadata: avicache.ServiceMetadataObj{
+					NamespaceServiceName: option.ServiceMetadata.NamespaceServiceName,
+				},
+			})
+
+			// assuming 1 IP per gateway
+			gwStatus := gw.Status.DeepCopy()
+			gwStatus.Addresses = []advl4v1alpha1pre1.GatewayAddress{{
+				Value: option.Vip,
+				Type:  advl4v1alpha1pre1.IPAddressType,
+			}}
+
+			// when statuses are synced during bootup
+			InitializeGatewayConditions(gwStatus, &gw.Spec, true)
+			UpdateGatewayStatusGWCondition(gwStatus, &UpdateGWStatusConditionOptions{
```

Comment:
if you update this to the Gateway and then call the UpdateGatewayStatusObject, it would always result into a diff and hence `DeepEqual` inside the UpdateGatewayStatusObject method will not be useful anymore. Hence I see that we are trying to update the Gateway status on every AKO boot up.